### PR TITLE
[GraphQL] Different security and serialization for item_query and collection_query

### PR DESCRIPTION
**Pull request by mahmoodbazdar**
Updating docs for this pull request:
https://github.com/api-platform/core/pull/3015

### CODE REVIEW

File: core/graphql.md [hunk 1]
```diff
@@ -864,12 +867,12 @@ class Book
     // ...
 
     /**
-     * @Groups({"read", "write", "query"})
+     * @Groups({"read", "write", "item-query","collection-query"})
```
Comment (alanpoulain) [inline core/graphql.md]:
```suggestion
     * @Groups({"read", "write", "item-query", "collection_query"})
```

File: core/graphql.md [hunk 2]
```diff
@@ -864,12 +867,12 @@ class Book
...
      */
     public $name;
 
     /**
-     * @Groups({"read", "mutation"})
+     * @Groups({"read", "mutation","collection-query"})
```
Comment (alanpoulain) [inline core/graphql.md]:
```suggestion
     * @Groups({"read", "mutation", "collection_query"})
```

File: core/graphql.md [hunk 3]
```diff
@@ -879,7 +882,9 @@ class Book
 
 In this case, the REST endpoint will be able to get the two attributes of the book and to modify only its name.
 
-The GraphQL endpoint will be able to query only the name. It will only be able to create a book with an author.
+The GraphQL item endpoint will be able to query only the name.
```
Comment (alanpoulain) [inline core/graphql.md]:
```suggestion
The GraphQL endpoint will be able to query only the name of an item.
```

File: core/graphql.md [hunk 4]
```diff
@@ -879,7 +882,9 @@ class Book
...
+The GraphQL collection endpoint will be able to query the name and author.
```
Comment (alanpoulain) [inline core/graphql.md]:
```suggestion
It will be able to query the name and author of the collection.
```

**Comment by alanpoulain** [comment]
Thanks! I would have used snake_case everywhere.
It would be more logical if you could have more details when querying an item and less when querying a collection don't you think?

**Comment by mahmoodbazdar** [comment]
Yes you are right, I'll change it.

File: core/graphql.md [hunk 5]
```diff
@@ -879,7 +882,9 @@ class Book
 
 In this case, the REST endpoint will be able to get the two attributes of the book and to modify only its name.
 
-The GraphQL endpoint will be able to query only the name. It will only be able to create a book with an author.
+The GraphQL item endpoint will be able to query the name and author of an item.
```
Comment (alanpoulain) [inline core/graphql.md]:
```suggestion
The GraphQL endpoint will be able to query the name and author of an item.
```

**Pull request merged**